docs/dev/baremetal.md

```diff
@@ -35,6 +35,24 @@ Forking is painful, but ...
   openshift/installer, it's quite beneficial for us to be working in
   that codebase from the start.
 
+## What's next?
+
+### Bootstrap Ignition Customizations
+
+We need to make are metal specific bootstrap ignition customizations.
+How best to do it? With per platform directories for assets?
+
+### Provisioning Master Nodes
+
+Do we use baremetal-operator to drive the provisioning? Do this via
+terrarorm or directly?
```

Comment:
I'm -1 on the operator since it seems like too big a deviation from the openshift-install infrastructure.

I'd lean toward directly, or via terraform (another variation on directly).  Neither seem straight forward.  The terraform path fits in much cleaner, but requires us to write a bunch of terraform provider code.  Direct seems easier, but is very different from the other platforms, which are all terraform, so that might be an ugly integration.

Reply:
I'm thinking about this a bit flippantly, honestly - whatever hack we were going to do in dev-scripts, let's hack that into kni-installer first. Like, instead of doing terraform.Apply() for the masters, spawn a shell script! No doubt the devil is in the details, I just mean we don't have to make the first cut of this closer perfect just because we're doing it golang.

Reply:
Let me know if the latest rev expresses that a bit better

Reply:
Yeah the gap here is just some way to drive Ironic to deploy the masters - support is going into gophercloud so we should be able to do this in golang, and potentially via Terraform, but for now I'm fine with a hack to get us started, atm we're just driving openstackclient via a script, which works.

Unless openshift-install is moving away from Terraform imminently, I think the right plan is to use terraform, which in theory should be simple-ish as there are far fewer resources to create than in the AWS/OpenStack case, but that assumes we can encapsulate the workflow to deploy nodes in terraform-provider-openstack which may or may not be acceptable to the maintainers (or, we create our own terraform provider which might be an easier first step)